Content.Server/GameObjects/Components/Projectiles/HitscanComponent.cs

```diff
@@ -0,0 +1,174 @@
+using System;
+using Content.Shared.GameObjects;
+using Content.Shared.Physics;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.EntitySystemMessages;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Serialization;
+using Robust.Shared.Interfaces.Timing;
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+using Robust.Shared.Physics;
+using Robust.Shared.Serialization;
+using Timer = Robust.Shared.Timers.Timer;
+
+namespace Content.Server.GameObjects.Components.Projectiles
+{
+    /// <summary>
+    /// Lasers etc.
+    /// </summary>
+    [RegisterComponent]
+    public class HitscanComponent : Component
+    {
+        public override string Name => "Hitscan";
+        public CollisionGroup CollisionMask => (CollisionGroup) _collisionMask;
+        private int _collisionMask;
+        
+        public float Damage
+        {
+            get => _damage;
+            set => _damage = value;
+        }
+        private float _damage;
+        public DamageType DamageType => _damageType;
+        private DamageType _damageType;
+        public float MaxLength => 20.0f;
+
+        private TimeSpan _startTime;
+        private TimeSpan _deathTime;
+
+        public float ColorModifier { get; set; } = 1.0f;
+        private string _spriteName;
+        private string _muzzleFlash;
+        private string _impactFlash;
+        private string _soundHitWall;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(ref _collisionMask, "layers", (int) CollisionGroup.Opaque, WithFormat.Flags<CollisionLayer>());
+            serializer.DataField(ref _damage, "damage", 10.0f);
+            serializer.DataField(ref _damageType, "damageType", DamageType.Heat);
+            serializer.DataField(ref _spriteName, "spriteName", "Objects/Guns/Projectiles/laser.png");
+            serializer.DataField(ref _muzzleFlash, "muzzleFlash", null);
+            serializer.DataField(ref _impactFlash, "impactFlash", null);
+            serializer.DataField(ref _soundHitWall, "soundHitWall", "/Audio/Guns/Hits/laser_sear_wall.ogg");
+        }
+
+        public void FireEffects(IEntity user, float distance, Angle angle, IEntity hitEntity = null)
+        {
+            var effectSystem = IoCManager.Resolve<IEntitySystemManager>().GetEntitySystem<EffectSystem>();
```

Comment:
You can replace this kind of call with `EntitySystem.Get`. It's shorter and more readable for the same effect.